gen.go

```diff
@@ -0,0 +1,215 @@
+package avro
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/token"
+	"io"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+type GenConf struct {
+	PackageName string
+	OutFileName string
+}
+
+var primitiveMappings = map[Type]string{
+	"string":  "string",
+	"bytes":   "[]byte",
+	"int":     "int",
+	"long":    "int64",
+	"float":   "float32",
+	"double":  "float64",
+	"boolean": "bool",
+}
+
+func GenerateFrom(gc GenConf, s string) (io.Reader, error) {
+	schema, err := Parse(s)
+	if err != nil {
+		return nil, err
+	}
+
+	rSchema, ok := schema.(*RecordSchema)
+	if !ok {
+		return nil, errors.New("can only generate Go code from Record Schemas")
+	}
+	file := ast.File{
+		Name: &ast.Ident{Name: strcase.ToSnake(gc.PackageName)},
+	}
+	generateFrom(rSchema, &file)
+
+	buf := &bytes.Buffer{}
+	if err = writeFile(buf, &file); err != nil {
+		return nil, err
+	}
+	return buf, nil
```

Comment:
Perhaps we could run `gofmt` on the output. Example in code here: https://gophersnippets.com/how-to-format-go-code-programmatically

Reply:
Thought about it and didn't want to jump the gun. I'm OK with it, I'll add it.